client/my-sites/plans/controller.jsx

```diff
@@ -34,7 +34,11 @@ export function plans( context, next ) {
 			withDiscount={ context.query.discount }
 			discountEndDate={ context.query.ts }
 			redirectTo={ context.query.redirect_to }
-			redirectToAddDomainFlow={ context.query.addDomainFlow }
+			redirectToAddDomainFlow={
+				context.query.addDomainFlow !== undefined
+					? context.query.addDomainFlow === 'true'
+					: undefined
+			}
```

Comment:
Not having `addDomainFlow` is different from it being `false`, that's why this check is done.

Reply:
When I looked at the back end patch, I was wondering if we should just either include the query string or not include it instead of setting the true/false value.  I guess this seems like a good way to handle it though.

Reply:
Yeah, I decided on keeping the parameter because we needed to differentiate between not having it (and showing the monthly/annual toggle) vs having it set as false (and hiding the monthly toggle + skipping the domain step) vs having it set as true (and hiding the monthly toggle + showing the domain step). It's a little complicated because it's actually 3 possibilities 😅

Reply:
You can also use this syntax

```suggestion
			redirectToAddDomainFlow={
				context.query.addDomainFlow ?? context.query.addDomainFlow
			}
```

Reply:
This is a nice suggestion 🙂 but I think it doesn't work in this case because we need to check if `addDomainFlow` is equal to string "true", and if it is, then the resulting value should be boolean true. We could do `addDomainFlow ?? addDomainFlow === 'true'`, but in that case the `undefined` value would never be used.

Reply:
Sorry for some reason I read `true` instead of `'true'.` Shame on me 🙃

Reply:
No worries, that's very easy to miss 😄